app/code/Magento/InventoryApi/Api/Data/ReservationInterface.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventoryApi\Api\Data;
+
+use Magento\Framework\Api\ExtensibleDataInterface;
+use Magento\InventoryApi\Api\Data\ReservationExtensionInterface;
+
+/**
+ * The entity responsible for reservations, created to keep inventory amount (product quantity) up-to-date.
+ * It is created to have a state between order creation and inventory deduction (deduction of specific SourceItems)
+ *
+ * @api
+ */
+interface ReservationInterface extends ExtensibleDataInterface
```

Comment:
Looks like we no need to expose Reservation API for WebAPI (REST and SOAP). 
Currently, in Magento 2 WebAPI imposes some restrictions for entity interfaces (existence getter and setter methods).
But because we would not expose reservations over web (REST , SOAP) - we could use any interface we want.
And because we agreed that Reservation is append-only **immutable entity** we could eliminate all the setter methods.

So, we will end-up with ReservationInterface consisting of just getter methods.
And we need to introduce ReservationBuilderInterface which will allow the possibility to set data into the reservation when we need to create one.
after that we could build Reservation entity.
```php
$reservationBuilder->setStockId(1);
$reservationBuilder->setSku('sku');
$reservationBuilder->setQty(10);
$newReservation = $reservationBuilder->build();
//now we could save Reservation entity 
$reservationAppend->execute([$newReservation]);
```

Doing so, we could ensure immutability on the level of Reservation interface